core/src/com/unciv/logic/automation/UnitAutomation.kt

```diff
@@ -182,15 +193,14 @@ class UnitAutomation {
         if (unit.type.isCivilian()) return false
         val tilesThatCanWalkToAndThenPillage = unitDistanceToTiles
                 .filter { it.value.totalDistance < unit.currentMovement }.keys
-                .filter { unit.movement.canMoveTo(it) && UnitActions().canPillage(unit, it) }
+                .filter { unit.movement.canMoveTo(it) && UnitActions.canPillage(unit, it) }
 
         if (tilesThatCanWalkToAndThenPillage.isEmpty()) return false
         val tileToPillage = tilesThatCanWalkToAndThenPillage.maxBy { it.getDefensiveBonus() }!!
         if (unit.getTile() != tileToPillage)
             unit.movement.moveToTile(tileToPillage)
 
-        UnitActions().getUnitActions(unit, UncivGame.Current.worldScreen)
-                .first { it.type == UnitActionType.Pillage }.action?.invoke()
+        UnitActions.getPillageAction(unit, tileToPillage)?.invoke()
```

Comment:
Again, directly get the Pillage action instead of grabbing all the unit actions list and filtering